### ISSUE TITLE
Error: Specified Type Does Not Refine Haskell Type for Data.ByteString.elemIndex

### ISSUE THREAD

**Issue by fhaust** [ucsd-progsys/liquidhaskell#924]
I compiled *Liquid Haskell* from commit a63ad626cfb81163c16f43b0f585f6331111b7f6 (giving me a kind `liquidhaskell-0.7.0.0` installation) . Converted an existing project to use `liquidhaskell-cabal` and just ran the build process. Liquid Haskell reports:

```
     /home/florian/Dokumente/Projects/Haskell/liquidhaskell/.stack-work/install/x86_64-linux/nightly-2016-05-21/7.10.3/share/x86_64-linux-ghc-7.10.3/liquidhaskell-0.7.0.0/include/Data/ByteString.spec:310:8: Error: Specified Type Does Not Refine Haskell Type for Data.ByteString.elemIndex
     
     310 |     :: Data.Word.Word8
                  ^
     
     The Liquid type
      
         GHC.Word.Word8
    -> Data.ByteString.Internal.ByteString
    -> GHC.Base.Maybe GHC.Types.Int elemIndices
      
     is inconsistent with the Haskell type
      
         GHC.Word.Word8
    -> Data.ByteString.Internal.ByteString
    -> GHC.Base.Maybe GHC.Types.Int
      
     defined at <no location info>
```

Not sure this is supposed to happen, `elemIndexEnd` is the first name from the [next non whitespace line](https://github.com/ucsd-progsys/liquidhaskell/blob/master/include/Data/ByteString.spec#L318). 

I am in the process of running `stack test` on the LH project and so far it hasn't reported any errors, so I would assume that a simple parsing bug is unlikely?

**Comment by ranjitjhala** [comment]
Ugh. I think it *is* a parsing bug -- @gridaphobe I thought this one was fixed a long time ago?

**Comment by ranjitjhala** [comment]
Can you check if changing line 304 in include/ByteString.spec to:

```haskell
index :: bs : Data.ByteString.ByteString -> { n : Int | 0 <= n && n < bslen bs } -> Data.Word.Word8
```

fixes it for now? (Or else post as a gist the file that is tickling this bug?)

**Comment by ranjitjhala** [comment]
related: 

https://github.com/ucsd-progsys/liquidhaskell/issues/707

**Comment by fhaust** [comment]
`stack test` finished successfully.

Changing that line does not help. Nor does changing the lines above and below it.

Here's some code that is triggering this: https://gist.github.com/fhaust/ff243f94cc4d977731ccb959b704540b

(That's the first part from here: http://chrisdone.com/posts/fast-haskell-c-parsing-xml )

**Comment by gridaphobe** [comment]
Oh dear, I thought we had fixed these issues too, but it's always been a bit hacky since the parser isn't properly indentation-sensitive.. Might be time to finally bite the bullet on that (see https://hackage.haskell.org/package/indents)

**Comment by fhaust** [comment]
Adding the parentheses (as mentioned in #707) does help.

**Comment by kosmikus** [comment]
Fixed via #1704.
